features/htcondor/client/service.pan

```diff
@@ -10,11 +10,9 @@ variable CONDOR_CONFIG = {
     # global & security mandatory
     file_list = list('global', 'security');
     foreach (i; file; file_list) {
-        SELF['cfgfiles'] = append(
-            dict(
-                'name', file,
-                'contents', 'features/htcondor/templ/' + file
-            )
+        SELF['cfgfiles'][length(SELF['cfgfiles'])] = dict(
```

Comment:
What was wrong with `append`?

Reply:
Good point. I do not know. If I try to use append (like in the original template) in this block I get 

`referenced variable of type nlist is not a list`

Any idea of why?

Reply:
I found the solution in the panc doc about **append**:
*The first variant takes a single argument and always operates on SELF. It will directly modify the value of SELF and give the modified list (SELF) as the return value... The second variant takes two arguments. The first argument is a list value, either a literal list value or a list calculated from a DML block. This version will create a copy of the given list and append the given value to the copy.* In our case, SELF is a dict, therefore we have the error. The solution  is to tell to the append function to which dict entry we should add the new dict:
```
        SELF['cfgfiles'] = append(SELF['cfgfiles'], dict(
            'name', file,
            'contents', 'features/htcondor/templ/' + file
        ));

```

Reply:
Thanks Jerome!